sw/device/lib/runtime/pmp.h

```diff
@@ -0,0 +1,312 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_RUNTIME_PMP_H_
+#define OPENTITAN_SW_DEVICE_LIB_RUNTIME_PMP_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+/**
+ * RVISC-V PMP address matching modes.
+ *
+ * PMP address matching modes, how protected regions are formed and
+ * implementation defined granularity information is described in:
+ * "Volume II: RISC-V Privileged Architectures V20190608-Priv-MSU-Ratified",
+ * "3.6.1 Physical Memory Protection CSRs", "Address Matching".
+ *
+ * PMP regions matching logic and prioritisation is described in the
+ * "Priority and Matching Logic" paragraph of the section "3.6.1".
+ *
+ * Note that granularity is implementation defined (default G=0, making the
+ * minimal PMP region size as 2^(G+2) = 4 bytes), and with G ≥ 1, the
+ * NA4 is not available. Granularity also determines the minimal region size in
+ * TOR and NAPOT modes.
+ */
+
+/**
+ * Attribute for functions which return errors that must be acknowledged.
+ *
+ * This attribute must be used to mark all PMP API that return an error value of
+ * some kind, to ensure that callers do not accidentally drop the error on the
+ * ground.
+ */
+#define PMP_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
+
+/**
+ * RV32 PMP CSR definitions.
+ *
+ * The values for these defines have been taken from the:
+ * "Volume II: RISC-V Privileged Architectures V20190608-Priv-MSU-Ratified",
+ * "3.6.1 Physical Memory Protection CSRs".
+ */
+#define PMP_REGIONS_NUM 16
```

Comment:
Do any of these absolutely need to be macros? If not, they should be constants.

Reply:
I don't mind converting them to constants, but I don't quite understand why?

Reply:
This is a prime example of something that can be done in a follow-up PR.